helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -305,8 +461,23 @@ public Object removeRoutingTableChangeListener(
    */
   public List<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName,
       String partitionName, String state) {
-    return _routingTableRef.get().getInstancesForResourceGroup(resourceGroupName, partitionName,
-        state);
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResourceGroup(resourceGroupName,
+        partitionName, state);
+  }
+
+  public List<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
Removed.